crmsh/bootstrap.py

```diff
@@ -2596,19 +2645,30 @@ def remove_qdevice():
     if not confirm("Removing QDevice service and configuration from cluster: Are you sure?"):
         return
 
+    utils.check_all_nodes_reachable()
+    _context.qdevice_reload_policy = evaluate_qdevice_quorum_effect(QDEVICE_REMOVE)
+
     status("Disable corosync-qdevice.service")
     invoke("crm cluster run 'systemctl disable corosync-qdevice'")
-    status("Stopping corosync-qdevice.service")
-    invoke("crm cluster run 'systemctl stop corosync-qdevice'")
+    if _context.qdevice_reload_policy == QdevicePolicy.QDEVICE_RELOAD:
+        status("Stopping corosync-qdevice.service")
+        invoke("crm cluster run 'systemctl stop corosync-qdevice'")
 
     status_long("Removing QDevice configuration from cluster")
     qnetd_host = corosync.get_value('quorum.device.net.host')
     qdevice_inst = corosync.QDevice(qnetd_host)
     qdevice_inst.remove_qdevice_config()
     qdevice_inst.remove_qdevice_db()
     update_expected_votes()
-    invoke("crm cluster run 'crm corosync reload'")
     status_done()
+    if _context.qdevice_reload_policy == QdevicePolicy.QDEVICE_RELOAD:
+        invoke("crm cluster run 'crm corosync reload'")
+    elif _context.qdevice_reload_policy == QdevicePolicy.QDEVICE_RESTART:
+        status_long("Restarting cluster service")
+        utils.cluster_run_cmd("crm cluster restart")
+        status_done()
```

Comment:
Is this `status _done()` duplicate given that there's one above? Maybe remove this one and move the common one to the end of the function?

Reply:
Not duplicated. Previous `status_done` was for previous `status_long`, to print `done` after `description...`